jwst/tso_photometry/tso_photometry.py

```diff
@@ -54,14 +60,30 @@ def tso_aperture_photometry(datamodel, xcenter, ycenter, radius, radius_inner, r
         phot_aper = CircularAperture((xcenter, ycenter), r=radius)
         bkg_aper = CircularAnnulus((xcenter, ycenter), r_in=radius_inner, r_out=radius_outer)
 
-    # convert the input data and errors from MJy/sr to Jy
-    if datamodel.meta.bunit_data != "MJy/sr":
-        raise ValueError("data is expected to be in units of MJy/sr")
-    factor = 1.0e6 * datamodel.meta.photometry.pixelarea_steradians
-    datamodel.data *= factor
-    datamodel.err *= factor
-    datamodel.meta.bunit_data = "Jy"
-    datamodel.meta.bunit_err = "Jy"
+    if datamodel.meta.bunit_data == "MJy/sr":
+        # Convert the input data and errors from MJy/sr to Jy
+        factor = 1e6 * datamodel.meta.photometry.pixelarea_steradians
+        datamodel.data *= factor
+        datamodel.err *= factor
+        datamodel.meta.bunit_data = "Jy"
+        datamodel.meta.bunit_err = "Jy"
+    elif datamodel.meta.bunit_data == "DN/s":
+        # Convert the input data and errors from DN/s to electrons
+        factor = datamodel.meta.exposure.integration_time * gain_model.data
+        datamodel.data *= factor
+        datamodel.err *= factor
+        datamodel.meta.bunit_data = "electron"
+        datamodel.meta.bunit_err = "electron"
+    elif datamodel.meta.bunit_data == "DN":
```

Comment:
I'm unclear why we need this option to catch input data of unit 'DN'?  Isn't the input option either 'MJy/sr' in regular usage, or 'DN/s' if the photom step was skipped?

Reply:
Ah, so this isn't actually relevant for JWST but was relevant for HST I believe - I'll remove this.